testsuite/libra-swarm/src/swarm.rs

```diff
@@ -307,14 +315,26 @@ impl LibraSwarm {
             dir: swarm_config_dir,
             nodes: HashMap::new(),
             config,
+            role: RoleType::Validator,
         })
     }
 
-    pub fn launch_attempt(
-        &mut self,
-        role: RoleType,
-        disable_logging: bool,
-    ) -> Result<(), SwarmLaunchFailure> {
+    pub fn launch(&mut self) {
+        let num_attempts = 5;
+        for _ in 0..num_attempts {
+            match self.launch_attempt(false) {
+                Ok(_) => {
```

Comment:
super nit: any reason we can't compact this if each action is only a singe line? Seems like a lot of curly braces :P